template/src/Autovis/Controls.js

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import "./controls.css";
+
+const convertSecToTime = secs => {
+  var h = Math.floor(secs / 3600);
+  var m = Math.floor((secs % 3600) / 60);
+  var s = Math.floor((secs % 3600) % 60);
+  return `${h < 10 ? "0" : ""}${h}:${m < 10 ? "0" : ""}${m}:${
+    s < 10 ? "0" : ""
+  }${s}`;
+};
+
+const convertPositionToSeconds = (e, totalSeconds) => {
+  const rect = e.currentTarget.getBoundingClientRect();
+  const mouseXRel = e.clientX - rect.x;
+  const seekRatio = Math.min(Math.max(0, mouseXRel / rect.width), 1);
+  const seconds = seekRatio * totalSeconds;
+  return seconds;
+};
+
+const convertcurrentTimeToSliderWidth = (currentSeconds, totalSeconds) => {
```

Comment:
this can be made a bit more general and broken out into its own file. maybe something like
`toPercent = (ratio) => ratio * 100 + '%'`

then down below:
`width: toPercent(currentTime/duration)`